pandas/core/generic.py

```diff
@@ -4522,6 +4530,102 @@ def copy(self, deep=True):
         Returns
         -------
         copy : type of caller
+
+        Examples
+        --------
+        >>> s = pd.Series([1,2], dtype="int32", index=["a", "b"])
+        >>> s
+        a    1
+        b    2
+        dtype: int32
+        >>> s_copy = s.copy()
+        >>> s_copy
+        a    1
+        b    2
+        dtype: int32
+
+        Shallow copy versus default (deep) copy:
+
+        In a shallow copy the index and data is a reference to the original
+        objects'.
+
+        >>> s = pd.Series([1,2], dtype="int32", index=["a", "b"])
+        >>> deep_copy = s.copy()
+        >>> shallow_copy = s.copy(deep=False)
+        >>> id(s) == id(shallow_copy)
+        False
+        >>> id(s.index) == id(shallow_copy.index)
+        True
+        >>> id(s.values) == id(shallow_copy.values)
+        True
+        >>> id(s) == id(deep_copy)
+        False
+        >>> id(s.index) == id(deep_copy.index)
+        False
+        >>> id(s.values) == id(deep_copy.values)
+        False
+        >>> s[0] = 3
+        >>> s
+        a    3
+        b    2
+        dtype: int32
+        >>> shallow_copy
+        a    3
+        b    2
+        dtype: int32
+        >>> deep_copy
+        a    1
+        b    2
+        dtype: int32
+        >>> shallow_copy[0] = 4
+        >>> s
+        a    4
+        b    2
+        dtype: int32
+        >>> shallow_copy
+        a    4
+        b    2
+        dtype: int32
+        >>> deep_copy
+        a    1
+        b    2
+        dtype: int32
+
+        When copying object containing python objects, deep copy will
+        copy the indices and data but will not do so recursively.
+
+        >>> class Point(object):
```

Comment:
Is it required to put a class in this docstring? Again for the sake of being concise wonder if you can just use lists to illustrate

Reply:
Thanks!  I made this change plus the other requested changes.